plugin.video.nhkworldtv/lib/episode.py

```diff
@@ -0,0 +1,305 @@
+        self._aired = local_date
+
+    @property
+    def broadcast_end_date(self):
+        """ Gets the current broadcast end date in local timezone """
+        return self._broadcast_end_date
+
+    @broadcast_end_date.setter
+    def broadcast_end_date(self, value):
+        """ Sets the current broadcast end date from the timestamp value """
+        timestamp = int(value) // 1000
+        local_date = utils.to_local_time(timestamp)
+        self._broadcast_end_date = local_date
+
+    @property
+    def thumb(self):
+        """ Gets the thumbnail URL """
+        return self._thumb
+
+    @thumb.setter
+    def thumb(self, value):
+        """ Sets thumbnail URL """
+        if ('/nhkworld/' in value and not self.absolute_image_url):
+            self._thumb = url.get_NHK_website_url(value)
+        else:
+            self._thumb = value
+
+    @property
+    def fanart(self):
+        """ Gets the fanart URL """
+        return self._fanart
+
+    @fanart.setter
+    def fanart(self, value):
+        """ Sets thumbnail URL """
+        if ('/nhkworld/' in value and not self.absolute_image_url):
+            self._fanart = url.get_NHK_website_url(value)
+        else:
+            self._fanart = value
+
+    @property
+    def date(self):
+        """ Get the sort date
+        Date Format: 01.01.2009
+        """
+        if (self._date is not None):
+            return self._date.strftime('%d.%m.%Y')
+        else:
+            return None
+
+    @property
+    def aired(self):
+        """ Get the aired date
+        Date Format: 2008-12-07
+        """
+        if (self._aired is not None):
+            return self._aired.strftime('%Y-%m-%d')
+        else:
+            return None
+
+    @property
+    def year(self):
+        if (self._date is not None):
+            self._year = self._date.strftime('%Y')
+            return self._year
+        else:
+            return None
+
+    @property
+    def video_info(self):
+        if (self._video_info is not None):
+            return self._video_info
+        elif (self.aspect is not None):
+            vi = {
+                'aspect': self.aspect,
+                'width': self.width,
+                'height': self.height
+            }
+            self._video_info = vi
+            return vi
+
+        else:
+            return None
+
+    @video_info.setter
+    def video_info(self, value):
+        """ Sets a Kodi Video Info """
+        self._video_info = value
+
+    @property
+    def cast(self):
+        """ Gets a cast info label
+        If it has not been set before, a default one will
+        be created from other fields
+        """
+        if (self._cast is not None):
+            return self._cast
+
+    @cast.setter
+    def cast(self, value):
+        """ Sets the cast info label """
+        self._cast = value
+
+    @property
+    def kodi_list_item(self):
+        """ Gets the current Kodi List Item """
+
+        if (self.url is not None):
```

Comment:
You might want to define offscreen=True to speed things up. See https://codedocs.xyz/xbmc/xbmc/group__python__xbmcgui__listitem.html#ga15bcb930a4f91dd97f8ed7c633a20618 . This change is just a recommendation.

Reply:
Thank you for the advice - I have changed the code.